modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -524,6 +524,64 @@ std::shared_ptr<DataFrame> combineDataFrames(std::vector<std::shared_ptr<DataFra
     return newDataFrame;
 }
 
+#include <warn/push>
+#include <warn/ignore/conversion>
+std::vector<std::uint32_t> filteredRows(const Column& col, dataframefilters::Filters filters) {
+
+    if (col.getColumnType() == ColumnType::Categorical) {
+        const auto& catCol = dynamic_cast<const CategoricalColumn&>(col);
+        std::vector<std::uint32_t> rows;
+        for (auto&& [row, value] : util::enumerate<std::uint32_t>(catCol.getValues())) {
+            auto test = util::overloaded{
+                [v = value](const std::function<bool(std::string_view)>& func) { return func(v); },
+                [](const auto&) { return false; }};
+            if (std::any_of(filters.include.begin(), filters.include.end(),
+                            [&](const auto& f) { return std::visit(test, f.filter); })) {
+                rows.push_back(row);
+            }
+        }
+        return rows;
+    } else {
+        return col.getBuffer()
+            ->getRepresentation<BufferRAM>()
+            ->dispatch<std::vector<std::uint32_t>, dispatching::filter::Scalars>(
+                [filters](auto typedBuf) {
+                    using ValueType = util::PrecisionValueType<decltype(typedBuf)>;
+                    std::vector<std::uint32_t> rows;
```

Comment:
wonder how to best adopt this to multiple columns? Might want to pass the rows and loop over both the rows and the column rows in "lockstep", somehow. If a row is already discarded from one column we don't need to test that row in other columns?

Reply:
Either that or filtering each column on its own in combination with a bitset.

Reply:
Think it would make sense to take advantage of the fact that rows is sorted... probably want to use two row vectors and pingponging when doing multiple columns...

Reply:
Potentially. Could use the row set/BitSet of the previous column to avoid unnecessary tests. Though then there is the trade-off between running a test vs. checking the row set for existence.